src/pyodide-py/pyodide/_base.py

```diff
@@ -42,6 +43,31 @@ def open_url(url: str) -> StringIO:
     return StringIO(req.response)
 
 
+def quiet_code(code: str) -> bool:
+    """
+    Does the last nonwhitespace character of code is a semicolon?
+
+    This can be overridden to customize the way eval_code is silenced.
+    """
+    # largely inspired from IPython:
+    # https://github.com/ipython/ipython/blob/86d24741188b0cedd78ab080d498e775ed0e5272/IPython/core/displayhook.py#L84
+
+    codeio = StringIO(code)
```

Comment:
```suggestion
    # We need to wrap tokens in a buffer because:
    # "Tokenize requires one argument, readline, which must be a callable object which provides the same interface as the io.IOBase.readline() method of file objects"
    codeio = StringIO(code)
```

Reply:
Would have been friendly of the `tokenize` package to have an api that accepts a normal iterable.

Reply:
You're right!